config/jobs/kubernetes/sig-node/containerd.yaml

```diff
@@ -470,7 +470,7 @@ periodics:
       # Note: The GCE Node image used may have a dependency on the nvidia-driver-installer image defined in https://github.com/kubernetes/kubernetes/blob/master/test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml
       # If updating the image defined here, the cos-gpu-installer image may need to updated to support the corresponding COS image.
       - --env=KUBE_NODE_OS_DISTRIBUTION=gci
-      - --env=KUBE_GCE_NODE_IMAGE=cos-109-17800-0-47
+      - --env=KUBE_GCE_NODE_IMAGE=cos-beta-109-17800-66-33
```

Comment:
This cos version will not work as it still uses 2.35.

```
akhil@akhil-ThinkPad-L14:~ $ gsutil cat gs://cos-tools/17800.66.33/cos-package-info.json | jq '.buildTimePackages[] | select(.name=="glibc")'
{
  "category": "sys-libs",
  "name": "glibc",
  "version": "17800.66.33",
  "ebuild_version": "2.35-r21"
}
```

Reply:
yeah, other jobs(e.g. https://testgrid.k8s.io/sig-node-containerd#cos-cgroupv1-containerd-e2e) are also failing for the same reason. It's strange as I've seen containerd running when I tested them in my gc environment.

Reply:
>  It's strange as I've seen containerd running when I tested them in my gc e

How did you get the containerd binary? Downloaded from the containerd release page? or from the gcloud storage bucket which gets updated after every merge?

The gcloud bucket has binaries built in the kubekins-e2e image. I think we need to change how images are built in the post-submits

Reply:
I run the job, then ssh-ed to the gc instance and run `ps aux | grep containerd`. I could be that it was from the cos-beta image though. I'll try to reproduce it again.

One thing that I noticed is that ci-cos-cgroupv2-inplace-pod-resize-containerd-main-e2e-gce-serial  and ci-cos-cgroupv1-inplace-pod-resize-containerd-main-e2e-gce-serial use different containerd versions. The former uses containerd 1.7 and it works and the latter uses containerd 2.0 beta and it has GLIBC issues as well as other failing jobs. I think they should be using the same containerd version (2.0 beta from the main containerd branch).

Reply:
both jobs download containerd, but different versions, e.g. https://storage.googleapis.com/kubernetes-jenkins/logs/ci-cos-cgroupv2-inplace-pod-resize-containerd-main-e2e-gce-serial/1704950578846633984/artifacts/bootstrap-e2e-master/containerd-installation.log:
```
Sep 21 20:12:47.283912 bootstrap-e2e-master configure.sh[1126]: + version=1.7.0-770-g3ebe5d1c5
Sep 21 20:12:47.284146 bootstrap-e2e-master configure.sh[1126]: + TARBALL_GCS_NAME=containerd-cni-1.7.0-770-g3ebe5d1c5.linux-amd64.tar.gz
Sep 21 20:12:47.284254 bootstrap-e2e-master configure.sh[1126]: + TARBALL_GCS_PATH=https://storage.googleapis.com/k8s-staging-cri-tools/containerd/main/containerd-cni-1.7.0-770-g3ebe5d1c5.linux-amd64.tar.gz
Sep 21 20:12:47.284324 bootstrap-e2e-master configure.sh[1126]: + TARBALL=containerd.tar.gz
Sep 21 20:12:47.284392 bootstrap-e2e-master configure.sh[1126]: + tar_sha1=
Sep 21 20:12:47.284583 bootstrap-e2e-master configure.sh[1126]: + '[' -z 1.7.0-770-g3ebe5d1c5 ']'
Sep 21 20:12:47.285126 bootstrap-e2e-master configure.sh[1156]: ++ uname -m
Sep 21 20:12:47.286563 bootstrap-e2e-master configure.sh[1126]: + '[' x86_64 == aarch64 ']'
Sep 21 20:12:47.286682 bootstrap-e2e-master configure.sh[1126]: + is_preloaded containerd-cni-1.7.0-770-g3ebe5d1c5.linux-amd64.tar.gz ''
Sep 21 20:12:47.286795 bootstrap-e2e-master configure.sh[1126]: + local -r tar=containerd-cni-1.7.0-770-g3ebe5d1c5.linux-amd64.tar.gz
Sep 21 20:12:47.286904 bootstrap-e2e-master configure.sh[1126]: + local -r sha1=
Sep 21 20:12:47.286977 bootstrap-e2e-master configure.sh[1126]: + grep -qs containerd-cni-1.7.0-770-g3ebe5d1c5.linux-amd64.tar.gz, /home/kubernetes/preload_info
Sep 21 20:12:47.288550 bootstrap-e2e-master configure.sh[1158]: ++ set +x
Sep 21 20:12:47.898782 bootstrap-e2e-master configure.sh[1126]: + tar xvf containerd.tar.gz
Sep 21 20:12:47.910826 bootstrap-e2e-master configure.sh[1160]: cri-containerd.DEPRECATED.txt
Sep 21 20:12:47.911211 bootstrap-e2e-master configure.sh[1160]: etc/
Sep 21 20:12:47.911338 bootstrap-e2e-master configure.sh[1160]: etc/systemd/
Sep 21 20:12:47.911419 bootstrap-e2e-master configure.sh[1160]: etc/systemd/system/
Sep 21 20:12:47.911495 bootstrap-e2e-master configure.sh[1160]: etc/systemd/system/containerd.service
Sep 21 20:12:47.911591 bootstrap-e2e-master configure.sh[1160]: etc/cni/
Sep 21 20:12:47.911718 bootstrap-e2e-master configure.sh[1160]: etc/cni/net.d/
Sep 21 20:12:47.911835 bootstrap-e2e-master configure.sh[1160]: etc/cni/net.d/10-containerd-net.conflist
Sep 21 20:12:47.911952 bootstrap-e2e-master configure.sh[1160]: etc/crictl.yaml
Sep 21 20:12:47.912074 bootstrap-e2e-master configure.sh[1160]: usr/
Sep 21 20:12:47.912168 bootstrap-e2e-master configure.sh[1160]: usr/local/
Sep 21 20:12:47.912243 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/
Sep 21 20:12:47.912328 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/containerd-shim-runc-v2
Sep 21 20:12:48.041181 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/containerd-stress
Sep 21 20:12:48.293286 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/ctd-decoder
Sep 21 20:12:48.546381 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/ctr
Sep 21 20:12:48.808307 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/critest
Sep 21 20:12:49.337291 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/containerd
Sep 21 20:12:49.834706 bootstrap-e2e-master configure.sh[1160]: usr/local/bin/crictl
Sep 21 20:12:50.349827 bootstrap-e2e-master configure.sh[1160]: usr/local/sbin/
Sep 21 20:12:50.350306 bootstrap-e2e-master configure.sh[1160]: usr/local/sbin/runc
...
```
The question is why they're downloading different versions.

Reply:
@akhilerm 
> The question is why they're downloading different versions.

I hope I found the reason. The job run I was looking for (the latest!) was too old. My guess is at that time containerd 1.7 was still in main branch or something in the job config was pointing to it. I don't understand why the job is not running since September this year, but that's another story.